src/txmempool.cpp

```diff
@@ -635,7 +635,6 @@ void CTxMemPool::check(CChainState& active_chainstate) const
     uint64_t innerUsage = 0;
 
     CCoinsViewCache& active_coins_tip = active_chainstate.CoinsTip();
-    assert(std::addressof(::ChainstateActive().CoinsTip()) == std::addressof(active_coins_tip)); // TODO: REVIEW-ONLY, REMOVE IN FUTURE COMMIT
```

Comment:
23b5ebd02e584c7bb982ccbc28bbabd927b9c57e: I thought the background chain didn't have a mempool? Or it had a dummy mempool. I guess it's fine to remove this assert though.

Reply:
Same as above; called during unittests.